services/update.py

```diff
@@ -243,47 +190,62 @@ def get_update(self):
 
         return False
 
-    def get_bad_rdf(self):
-        return bad_rdf
-
-    def get_rdf(self):
+    def get_output(self):
         if self.is_valid():
             if self.get_update():
-                rdf = self.get_good_rdf()
+                output = self.get_success_output()
             else:
-                rdf = self.get_no_updates_rdf()
+                output = self.get_no_updates_output()
         else:
-            rdf = self.get_bad_rdf()
+            output = self.get_error_output()
         self.cursor.close()
         if self.conn:
             self.conn.close()
-        return rdf
+        return json.dumps(output)
 
-    def get_no_updates_rdf(self):
-        name = base.ADDON_SLUGS_UPDATE[self.data['type']]
-        return no_updates_rdf % ({'guid': self.data['guid'], 'type': name})
+    def get_error_output(self):
```

Comment:
Is this what Firefox expects for an error output?  The MDN doc was vague.  (It's a sensible error output, just wondering)

Reply:
I'm not sure what Firefox expects either. I went with what made the most sense, and we'll see if that works... tbh I don't think it matters much - Firefox doesn't stupidly crash if we return nonsense, worst case scenario it logs something to the browser console, which isn't a bad thing in this situation.